### PR TITLE
Work-order-history-doesn't-work-properly-with-pagination

### LINKED ISSUE #443: Work order history doesn't work properly with pagination
By determining uniques _after_ pagination, we may not get the right number per page. We should re-write this query so that we can distinct _before_ calling `Repo.paginate`

### PR DESCRIPTION
**Pull request by wanecode**
## Related issue

Fixes #443 Work order history doesn't work properly with pagination

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If needed, I've updated the changelog
- [x] Amber has QA'd this feature

### CODE REVIEW

**Comment by amberrignell** [comment]
Just one small comment - I think you forgot to update the number that's displayed

In the screenshot below, it says "11 to 20 of 16 results" when it should say "11 to 16"
Screenshot 2022-12-14 at 16 44 41

File: config/config.exs [hunk 1]
```diff
@@ -10,6 +10,8 @@ import Config
 config :lightning,
   ecto_repos: [Lightning.Repo]
 
+config :lightning, Lightning.Repo, types: Lightning.PostgrexTypes
```
Comment (stuartc) [inline config/config.exs:13]: @wanecode do we still need this since you used `nullif`?

**Review by stuartc — APPROVED** [review]
Other than my comment on adding PostgrexTypes this is good to me.

Please let me know if we still need it, and if not please remove it from `config.exs`.

File: config/config.exs (hunk 1 above)
Comment (wanecode) [reply config/config.exs:13]: @stuartc. Yes . the max clause needs infinity. Or we can hard a "max date"

**Comment by amberrignell** [comment]
Looks good, thanks Mamadou :)

**Pull request merged**